src/runtime/ggml/logging.cpp

```
@@ -13,49 +13,38 @@
             formatted_buffer, sizeof(formatted_buffer), "%s:%d:<%s> %s", file, line, func, buffer);
-        g_log_state.log_callback(level, formatted_buffer, g_log_state.log_callback_user_data);
+        ggml_log_callback callback;
+        void* user_data;
+        ggml_log_get(&callback, &user_data);
+        callback(level, formatted_buffer, user_data);
     } else {
         char* buffer2 = new char[len + 1];
-        vsnprintf(buffer2, len + 1, format, args);
+        vsnprintf(buffer2, len + 1, format, args_copy);
         buffer2[len] = 0;
         char formatted_buffer[4096];
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)